packages/harmony/src/components/button/FilterButton/FilterButton.tsx

```diff
@@ -0,0 +1,218 @@
+import {
+  forwardRef,
+  ForwardedRef,
+  RefObject,
+  useRef,
+  useState,
+  useCallback
+} from 'react'
+
+import { CSSObject, useTheme } from '@emotion/react'
+
+import { BaseButton } from 'components/button/BaseButton/BaseButton'
+import { Box, Flex, Paper, Popup } from 'components/layout'
+import { IconCaretDown, IconCloseAlt } from 'icons'
+
+import {
+  FilterButtonOption,
+  FilterButtonProps,
+  FilterButtonSize,
+  FilterButtonType
+} from '../types'
+
+export const FilterButton = forwardRef<HTMLButtonElement, FilterButtonProps>(
+  function FilterButton(
+    props: FilterButtonProps,
+    ref: ForwardedRef<HTMLButtonElement>
+  ) {
+    const {
+      label,
+      options,
+      onSelect,
+      variant = FilterButtonType.FILL_CONTAINER,
+      size = FilterButtonSize.DEFAULT,
+      iconRight = IconCaretDown,
+      popupAnchorOrigin,
+      popupTransformOrigin
```

Comment:
would there ever be a case where this button could come pre-filtered? in which case we would need to pass value in?

Reply:
Yeah good call there should be an initial value option. As far as it being fully "controlled" idts, but we can add if we see that happening.

Reply:
can be a part of the beta -> prod update

Reply:
i did it, np